### ISSUE TITLE
celt_fir5 could get some cleanup

### ISSUE THREAD

**Issue by est31** [xiph/opus#58]
The static function [celt_fir5](https://github.com/xiph/opus/blob/3fe744ea04fdcc418fb85c2c133d13372ebb019b/celt/pitch.c#L105-L144) is used only once, [at line 214](https://github.com/xiph/opus/blob/3fe744ea04fdcc418fb85c2c133d13372ebb019b/celt/pitch.c#L214) inside the `pitch_downsample` function. There are some minor issues that could warrant a cleanup:

* `mem` is sort of superflous. It is created on the stack of pitch_downsample and initialized with 0, and passed to celt_fir5 but there only the 0 values are read from mem, and while there is writing to mem afterwards, pitch_downsample never reads that. One could just simply initialize mem0-mem4 with 0 and drop the mem variable entirely, no?
* There is this weird aliasing happening between the x and y pointers; why not just simply have one pointer?

I might be missing something and this is all done to nudge the compiler to vectorize which it otherwise wouldn't do, but IMO if that's the case then it would be nice to have a comment in the code.

**Comment by jmvalin** [comment]
celt_fir5() just came from a more generic function that could do something useful with these args. Fixed now.
